### ISSUE TITLE
File Upload

### ISSUE THREAD

**Issue by Pd1r** [dotCMS/core#17796]
### Describe the bug
Upload jsp files to control the target server

### Steps to reproduce the behavior:
1. uri.startsWith () determines whether uri starts with / asset

![image](https://user-images.githubusercontent.com/30740738/71956154-5174a880-3225-11ea-8ddd-8592da8ce9fd.png)

2. Can bypass restricted access to files under assets, like 
/asdasd/../asset

![image](https://user-images.githubusercontent.com/30740738/71956305-bfb96b00-3225-11ea-9eb0-0f23e447868b.png)

![image](https://user-images.githubusercontent.com/30740738/71956312-c5af4c00-3225-11ea-8b1c-490fe3abcacf.png)

3. Upload malicious JSP file here

![image](https://user-images.githubusercontent.com/30740738/71956432-1c1c8a80-3226-11ea-9ac6-910ac0af1525.png)

4. Get file id
![image](https://user-images.githubusercontent.com/30740738/71956791-03f93b00-3227-11ea-9071-a7981bf7b287.png)

5. Execute arbitrary server commands
![image](https://user-images.githubusercontent.com/30740738/71956812-12475700-3227-11ea-9146-d8c4c478cbe3.png)

6. Can upload even without authorization
![image](https://user-images.githubusercontent.com/30740738/71956844-27bc8100-3227-11ea-8be8-f8de2e9c4246.png)

dir like this
![image](https://user-images.githubusercontent.com/30740738/71957007-90a3f900-3227-11ea-9030-e060be68ea87.png)

**Comment by wezell** [comment]
@Pd1r  Questions on this:
1. What app server is this running?  Tomcat or something else?  Is it behind any proxy or using any special connectors?
2. What OS is the server running?  Is it containerized?  If so, what is the base os.  If Linux, what distro?

I am trying to reproduce this and I cannot:
```
curl -XPOST http://localhost:8080/234aa/../assets/messages/cms_language_en.properties
```

**Comment by Pd1r** [comment]
server : tomcat 8.5.32
os : 10.14.6
Tool : Can't use curl, need Burpsuite 
Unreachable when I use localhost, It is possible to use the IP assigned by the router .

I downloaded from here
![image](https://user-images.githubusercontent.com/30740738/72034870-3a8b9000-32d1-11ea-958e-580e40bd81cf.png)
![image](https://user-images.githubusercontent.com/30740738/72034958-77f01d80-32d1-11ea-9064-35aa66d44abc.png)

![image](https://user-images.githubusercontent.com/30740738/72035285-68250900-32d2-11ea-821d-0fb503f10a46.png)

**Comment by wezell** [comment]
@Pd1r thank you for the report and details, I can confirm this.  We are working on a fix.

**Comment by jgambarios** [comment]
PR: https://github.com/dotCMS/core/pull/17809

**Comment by bryanboza** [comment]
Fixed, tested on release-5.2.4 // Postgres // FF

**Comment by cfi-gb** [comment]
> I am trying to reproduce this and I cannot:
>
> curl -XPOST http://localhost:8080/234aa/../assets/messages/cms_language_en.properties

Note that you need to pass the `--path-as-is` parameter for directory traversals in such curk calls:

> --path-as-is    Do not squash .. sequences in URL path
